yarn-project/sequencer-client/src/publisher/sequencer-publisher.ts

```diff
@@ -589,6 +595,12 @@ export class SequencerPublisher {
       return false;
     }
 
+    const payloadCode = await this.l1TxUtils.getCode(payload);
+    if (payloadCode === undefined || payloadCode === `0x` || payloadCode === `0x0`) {
```

Comment:
Can this be moved to a helper function, something like `isCodeEmpty`?
I'd also add a reference to the docs (or wherever) you found empty payload indicators (i.e., how do we _know_ `0x` represents empty payload, and e.g. `0x00` doesn't)

Reply:
I don't feel too strongly about this - meaning I'm fine with keeping the `if` statement here 🙂
The main reason I suggested moving this to a function is to give _semantics_ to this `if` and make it easier to extend and test in the future if needed (and not code (de)duplication)

---

By giving "semantic meaning"  - I was referring to a pattern usually called "named if", of which I'm a huge fan.